src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] dataBase;
+    private int size;
+
+    public ArrayList() {
+        this.dataBase = new Object[DEFAULT_CAPACITY];
+    }
+
+    private void grow(int minCapacity) {
+        if (size >= dataBase.length) {
+            int newCapacity = dataBase.length + (dataBase.length >> 1);
+            Object[] newDataBase = new Object[newCapacity];
+            System.arraycopy(dataBase, 0, newDataBase, 0, size);
+            dataBase = newDataBase;
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        grow(size + 1);
+        dataBase[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Entered invalid index, try again !");
+        }
```

Comment:
```suggestion
        if (index < 0 || index > size) {
            throw new ArrayListIndexOutOfBoundsException("Entered invalid index, try again !");
        }
```

move this check to the separate method

Reply:
it has no sense cuz in that case im using "index > size" and in other - "index >= size" otherwise it doesnt work, But i did it to other cases